public_html/assets/js/aws-s3-explorer.js

```diff
@@ -0,0 +1,664 @@
+            const map = {
+                '&': '&amp;',
+                '<': '&lt;',
+                '>': '&gt;',
+                '"': '&quot;',
+                "'": '&#x27;',
+                "`": '&grave;',
+                "/": '&#x2F;',
+            };
+            const reg = /[&<>"'`/]/ig;
+            return string.replace(reg, (match) => (map[match]));
+        }
+
+    function isthisdocument(bucket, key) {
+    return key === "index.html";
+    }
+
+    function isfolder(path) {
+        return path.endsWith('/');
+    }
+    function haspreview(data) {
+        var extension = data.Key.substring(data.Key.lastIndexOf('.'),data.Key.length);
+        var preview = [".csv",".tsv",".out",".table",".txt",".html"].includes(extension) && data.Size<80000000;
+        return preview
+    }
+    // Convert cars/vw/golf.png to golf.png
+    function fullpath2filename(path) {
+        return sanitize_html(path.replace(/^.*[\\\/]/, ''));
+    }
+
+    // Convert cars/vw/golf.png to cars/vw
+    function fullpath2pathname(path) {
+        return sanitize_html(path.substring(0, path.lastIndexOf('/')));
+    }
+
+    // Convert cars/vw/ to vw/
+    function prefix2folder(prefix) {
+        var parts = prefix.split('/');
+        return sanitize_html(parts[parts.length - 2] + '/');
+    }
+
+    // Remove hash from document URL
+    function removeHash() {
+        history.pushState("", document.title, window.location.pathname + window.location.search);
+    }
+
+    $("body").on("click",".preview-file-btn ",function(e){
+        var url = $(this).siblings('div').children('a')[0].href;
+        var extension = url.substring(url.lastIndexOf('.'),url.length);
+        var split_url = url.split('/');
+        var filename = split_url.slice(-1)[0];
+        var s3_url = "s3://"+ split_url[2].split(".")[0] + "/" + split_url.slice(3).join("/");
+
+        fetch(url)
+            .then(
+                function(response) {
+                    if (response.status !== 200) {
+                        console.log('Looks like there was a problem. Status Code: ' + response.status);
+                        return;
+                    }
+                    // Examine the text in the response
+                    response.text().then(function(data) {
+                    if(extension !== '.html'){
+                        data = '<pre><code>'+data+'</code></pre>';
+                    }
+                    else {
+                        data = '<iframe srcdoc="'+sanitize_html(data)+'" style="border:none; width:100%; height:1000px;"></iframe>';
+                    }
+                    var btn_group = `<div class="btn-group" role="group">
+                      <button type="button" class="btn btn-outline-secondary copy-url" data-target=${url}>Copy URL</button>
+                      <button type="button" class="btn btn-outline-secondary copy-url" data-target=${s3_url}>Copy S3 URL</button>
+                    </div>`;
```

Comment:
Additional button to just open in a new tab would be nice:

```suggestion
                    var btn_group = `<div class="btn-group" role="group">
                      <a class="btn btn-outline-secondary" href="${url}" target="_blank">Open in new tab</a>
                      <button type="button" class="btn btn-outline-secondary copy-url" data-target=${url}>Copy URL</button>
                      <button type="button" class="btn btn-outline-secondary copy-url" data-target=${s3_url}>Copy S3 URL</button>
                    </div>`;
```